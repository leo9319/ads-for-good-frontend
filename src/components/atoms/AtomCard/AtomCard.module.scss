.wrapper {
  display: flex;
  background-color: transparent;

  .card {
    border-radius: var(--cardBorderRadius);
    padding: var(--cardPadding);
    box-shadow: 0 4px 10px rgba(0, 0, 0, 0.1);
    box-sizing: border-box;
    height: 100%;

    .innerBox {
      height: 100%;
      overflow: hidden;

      > * {
        width: 100%;
        height: 100%;
        border: none;
      }
    }
  }
}
