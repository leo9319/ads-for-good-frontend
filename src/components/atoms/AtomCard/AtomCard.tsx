import React, { ReactElement } from 'react';
import { classNames } from '@utils/common/classNames';
import { getStylesAsCssProperties } from '@utils/common/styles';

import styles from './AtomCard.module.scss';
import { Flex } from '@radix-ui/themes';

/**
 * @interface
 * The Card props type of {@link AtomCard}.
 */
export interface AtomCardProps {
  /**
   * Passing a Children as react element
   * @property
   */
  children?: React.ReactNode;
  /**
   * Size of the card
   * @property
   */
  size?: '1' | '2' | '3';
  /**
   * Additional class name for the card
   * @property
   */
  className?: string;
}

const sizeStyles = {
  1: {
    padding: '80px',
    borderRadius: '32px',
  },
  2: {
    padding: '40px',
    borderRadius: '32px',
  },
  3: {
    padding: '16px',
    borderRadius: '16px',
  },
};

/**
 *
 * @group Atoms
 *
 * @category Component
 *
 * Atom Card component is a wrapper for content with a background and padding.
 * It can be used to create a card-like UI element.
 *
 * @param {AtomCardProps} props - The props for the Card component.
 *
 * @returns {JSX.Element} - The Card component.
 *
 */
export const AtomCard = ({
  size = '2',
  className,
  children,
}: AtomCardProps): ReactElement => {
  const { padding, borderRadius } = sizeStyles[size];
  return (
    <Flex
      display="flex"
      justify="center"
      align="center"
      width="100%"
      height="100%"
      className={classNames(styles.wrapper, className)}
    >
      <Flex
        display="flex"
        justify="center"
        align="center"
        width="100%"
        className={styles.card}
        style={getStylesAsCssProperties({
          '--cardPadding': padding,
          '--cardBorderRadius': borderRadius,
        })}
      >
        <Flex
          display="flex"
          justify="center"
          align="center"
          width="100%"
          className={styles.innerBox}
        >
          {children}
        </Flex>
      </Flex>
    </Flex>
  );
};

export default AtomCard;
