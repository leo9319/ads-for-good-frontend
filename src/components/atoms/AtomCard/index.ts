export { default } from './AtomCard';
export { AtomCard } from './AtomCard';
export type { AtomCardProps } from './AtomCard';
