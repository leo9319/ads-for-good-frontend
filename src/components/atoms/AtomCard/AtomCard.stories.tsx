import React from 'react';
import type { Meta, StoryObj } from '@storybook/react';
import AtomCard from './AtomCard';
import { Theme } from '@radix-ui/themes';

const meta = {
  title: 'Library/Atoms/AtomCard',
  component: AtomCard,
  parameters: {
    layout: 'fullscreen',
  },
  tags: ['autodocs'],
  decorators: [
    Story => (
      <Theme
        style={{
          width: '100vw',
          height: '100vh',
          backgroundColor: 'white',
        }}
      >
        <Story />
      </Theme>
    ),
  ],
  argTypes: {
    children: {
      description: 'React JS Children of the card',
      control: { type: 'object' },
    },
    size: {
      description: 'Size of the card?',
      control: 'select',
      options: ['1', '2', '3'],
    },
    className: {
      description: 'className of the card to be passed',
      control: { type: 'text' },
    },
  },
} satisfies Meta<typeof AtomCard>;

export default meta;
type Story = StoryObj<typeof meta>;

export const Default: Story = {
  args: {
    size: '2',
    children: (
      <div
        style={{
          backgroundColor: '#fde6e8',
          display: 'flex',
          justifyContent: 'center',
          alignItems: 'center',
        }}
      >
        Card 1
      </div>
    ),
  },
};
