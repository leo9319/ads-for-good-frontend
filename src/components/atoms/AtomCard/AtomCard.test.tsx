import React from 'react';
import AtomCard from './AtomCard';
import { render } from '@testing-library/react';

describe('Atom Card', () => {
  it('should render successfully', () => {
    const { baseElement } = render(
      <AtomCard>
        <div>Card 1</div>
      </AtomCard>
    );
    expect(baseElement).toMatchSnapshot();
  });
});
