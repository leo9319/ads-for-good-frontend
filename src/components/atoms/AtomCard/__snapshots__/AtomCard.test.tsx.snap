// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Atom Card should render successfully 1`] = `
<body>
  <div>
    <div
      class="rt-Flex rt-r-display-flex rt-r-ai-center rt-r-jc-center rt-r-w rt-r-h wrapper"
      style="--width: 100%; --height: 100%;"
    >
      <div
        class="rt-Flex rt-r-display-flex rt-r-ai-center rt-r-jc-center rt-r-w card"
        style="--width: 100%; --cardPadding: 40px; --cardBorderRadius: 32px;"
      >
        <div
          class="rt-Flex rt-r-display-flex rt-r-ai-center rt-r-jc-center rt-r-w innerBox"
          style="--width: 100%;"
        >
          <div>
            Card 1
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;
